pkg/controller/storagecluster/storagecluster.go

```diff
@@ -242,6 +243,8 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
+	c.registerCSRAutoApproval(cluster)
```

Comment:
I'm not sure if this is the best place to do this? probably after waiting for migration approval? idk

Reply:
I had to find a way how to bind the `StorageCluster`-controller updates, with CSR-controller
* note that StorageCluster YAML carries `disable-csr-approve` annotation that turns CSR-approval -controller on or off for that particular ClusterID
* so, we register the callback here, to configure the CSR-controller